src/core/public/legacy_platform/legacy_platform_service.test.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const mockLoadOrder: string[] = [];
+
+const mockUiMetadataInit = jest.fn();
+jest.mock('ui/metadata', () => {
+  mockLoadOrder.push('ui/metadata');
+  return {
+    __newPlatformInit__: mockUiMetadataInit,
+  };
+});
+
+const mockUiChromeBootstrap = jest.fn();
+jest.mock('ui/chrome', () => {
+  mockLoadOrder.push('ui/chrome');
+  return {
+    bootstrap: mockUiChromeBootstrap,
+  };
+});
+
+const mockUiTestHarnessBootstrap = jest.fn();
+jest.mock('ui/test_harness', () => {
+  mockLoadOrder.push('ui/test_harness');
+  return {
+    bootstrap: mockUiTestHarnessBootstrap,
+  };
+});
+
+import { LegacyPlatformService } from './legacy_platform_service';
+
+const injectedMetadataStartContract = {
+  getLegacyMetadata: jest.fn(),
+};
+
+const defaultParams = {
+  rootDomElement: { someDomElement: true } as any,
+  requireLegacyFiles: jest.fn(() => {
+    mockLoadOrder.push('legacy files');
+  }),
+};
+
+beforeEach(() => {
```

Comment:
__question:__ why `beforeEach` and not `afterEach`? Not sure how it works in Jest exactly, but in the past and other test frameworks cleanups in the `beforeEach` masked tests that were executed before this test suite and didn't do a proper cleanup.